Sources/NIOHTTP2/HTTP2PipelineHelpers.swift

```diff
@@ -220,103 +220,6 @@ extension Channel {
         }
     }
 
-    /// Configures a `ChannelPipeline` to speak HTTP/2 and sets up mapping functions so that it may be interacted with from concurrent code.
-    ///
-    /// In general this is not entirely useful by itself, as HTTP/2 is a negotiated protocol. This helper does not handle negotiation.
-    /// Instead, this simply adds the handler required to speak HTTP/2 after negotiation has completed, or when agreed by prior knowledge.
-    /// Use this function to setup a HTTP/2 pipeline if you wish to use async sequence abstractions over inbound and outbound streams.
-    /// Using this rather than implementing a similar function yourself allows that pipeline to evolve without breaking your code.
-    ///
-    /// - Parameters:
-    ///   - mode: The mode this pipeline will operate in, server or client.
-    ///   - configuration: The settings that will be used when establishing the connection and new streams.
-    ///   - position: The position in the pipeline into which to insert this handler.
-    ///   - inboundStreamInitializer: A closure that will be called whenever the remote peer initiates a new stream.
-    /// - Returns: An `EventLoopFuture` containing the `AsyncStreamMultiplexer` inserted into this pipeline, which can
-    ///     be used to initiate new streams and iterate over inbound HTTP/2 stream channels.
-    @available(macOS 10.15, iOS 13.0, watchOS 6.0, tvOS 13.0, *)
-    @_spi(AsyncChannel)
-    public func configureAsyncHTTP2Pipeline<Output>(
-        mode: NIOHTTP2Handler.ParserMode,
-        configuration: NIOHTTP2Handler.Configuration = .init(),
-        position: ChannelPipeline.Position = .last,
-        inboundStreamInitializer: @escaping NIOChannelInitializerWithOutput<Output>
-    ) -> EventLoopFuture<NIOHTTP2Handler.AsyncStreamMultiplexer<Output>> {
-        if self.eventLoop.inEventLoop {
-            return self.eventLoop.makeCompletedFuture {
-                return try self.pipeline.syncOperations.configureAsyncHTTP2Pipeline(
-                    mode: mode,
-                    configuration: configuration,
-                    position: position,
-                    inboundStreamInitializer: inboundStreamInitializer
-                )
-            }
-        } else {
-            return self.eventLoop.submit {
-                return try self.pipeline.syncOperations.configureAsyncHTTP2Pipeline(
-                    mode: mode,
-                    configuration: configuration,
-                    position: position,
-                    inboundStreamInitializer: inboundStreamInitializer
-                )
-            }
-        }
-    }
-
-    /// Configures a `ChannelPipeline` to speak HTTP/2 and wraps any created inbound stream channels in `NIOAsyncChannel`s.
```

Comment:
Why are we removing this variant with this change? It's not related to protocol negotiation.

Reply:
It is not removed. Just moved if further down.

Reply:
This was intentional. While they are not protocol negotiation specific they were conveniences for `NIOAsyncChannel`. For now we have taken all of those conveniences out from `NIO` as well since people can just wrap in their initialisers themselves and it avoids a bunch of APIs for us.